web/js/duplicates.js

```diff
@@ -179,6 +179,43 @@
                 layers[i].setVisibility(false);
             }
         }
+
+        var defect_popup;
+
+        function close_defect_popup() {
+            if (!!defect_popup) {
+                fixmystreet.map.removePopup(defect_popup);
+                defect_popup.destroy();
+                defect_popup = null;
+            }
+        }
+
+        // Handle clicks on defect pins
+        var select_defect = new OpenLayers.Control.SelectFeature(
+            fixmystreet.markers,
+            {
+                hover: true,
+                clickFeature: function (feature) {
+                    close_defect_popup();
```

Comment:
There's quite a lot of duplication here and in `map-OpenLayers.js` - could it be factored together somehow? There's also the issue that this is quite Oxfordshire-specific but is in the main codebase. Having said that, I don't have a good suggestion for hooking in here from OCC JS because we're in a closed scope and not a function that the cobrand JS could override.

I think `fixmystreet.cobrand` would be available when this code is called - what about a cobrand check that then calls the factored-together code in the Oxfordshire JS?

Reply:
I decided to use jQuery events so that other cobrands can hook into the same places in the future if necessary.

Done in 0b58e43af4d7997f44e2b37f1fb1f3ac255d21b9